### PR TITLE
[jaeger-client-cpp] New port

### PR DESCRIPTION
**Pull request by xgcssch**
This is a port of the jaeger c++ client.
In combination with the opentracing port it gives application the ability to produce tracedata.

This pull request contains an version upgrade to the opentracing port, as the previous version produced a release build as the debug version. This makes any code using the debug version to crash.

### CODE REVIEW

File: ports/jaeger-client-cpp/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+# Copy generated files over to jaeger-client-cpp 
+file(GLOB IDL_SOURCE_FILES LIST_DIRECTORIES false ${IDL_SOURCE_DIR}/*)
+file(COPY ${IDL_SOURCE_FILES} DESTINATION ${SOURCE_PATH}/src/jaegertracing/thrift-gen)
+
+# Generate Jaeger client
+vcpkg_install_cmake()
+
+# Integrate Debug CMake file
+file(RENAME ${CURRENT_PACKAGES_DIR}/debug/share/jaeger-client-cpp/jaegertracingTargets-debug.cmake ${CURRENT_PACKAGES_DIR}/share/jaeger-client-cpp/jaegertracingTargets-debug.cmake)
```
Comment (NancyLi1013) [inline ports/jaeger-client-cpp/portfile.cmake]: Could you help make a description about why there is only `jaegertracingTargets-debug.cmake` ?

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

File: ports/jaeger-client-cpp/portfile.cmake (hunk 1 above)
Comment (xgcssch) [reply ports/jaeger-client-cpp/portfile.cmake]: Hello! I implemented the proposed changes. 
There ist an jaegertracingTargets-releas.cmake already in place. Only the debug version resides in a seperate debug subdirectory. So this statement is only to integrate the debug version at the shared location.
Best regards
Sönke

Comment (NancyLi1013) [reply ports/jaeger-client-cpp/portfile.cmake]: Thanks for your update.
You can use `vcpkg_fixup_cmake_targets()` to handle this file.

File: ports/jaeger-client-cpp/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+endforeach()
+
+# Save generated proxy/stub target directory
+set(IDL_SOURCE_DIR "${SOURCE_PATH}/data/gen-cpp")
+
+# Get jaeger-client-cpp from github
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO jaegertracing/jaeger-client-cpp
+    REF v0.5.0
```
[the remark this answers is not in the document]
Comment (xgcssch) [reply ports/jaeger-client-cpp/portfile.cmake]: Changed that and introduced vcpkg_fixup_cmake_targets()

**Comment by NancyLi1013** [comment]
Hi @xgcssch thanks for this PR.
I noticed that this port failed on `arm64-windows` triplet. Is this expected? If this port doesn't support this triplet, please add `vcpkg_fail_port_install(ON_ARCH "arm64")` to the top of `portfie.cmake`.
At the same time, update `jaeger-client-cpp:arm64-windows=fail` to [ci.baseline.txt](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt).
If not, please try to fix it.

**Comment by xgcssch** [comment]
> Hi @xgcssch thanks for this PR.
> I noticed that this port failed on `arm64-windows` triplet. Is this expected? If this port doesn't support this triplet, please add `vcpkg_fail_port_install(ON_ARCH "arm64")` to the top of `portfie.cmake`.
> At the same time, update `jaeger-client-cpp:arm64-windows=fail` to [ci.baseline.txt](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt).
> If not, please try to fix it.

Hello!
As i've no access to an arm machine, i added the `vcpkg_fail_port_install()` call.
For the change in ci.baseline.txt: in my fork this file does not exist.
Do you have any idea how to incorporate this change in the current pull request?
I haven't had such a situation before. 
Best regards Soenke

File: ports/opentracing/portfile.cmake [hunk 3]
```diff
@@ -16,8 +16,8 @@ endif()
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO opentracing/opentracing-cpp
-    REF cf9b9d5c26ef985af2213521a4f0701b7e715db2
-    SHA512 75b77781c075c6814bf4a81d793e872ca47447fe82a4cad878bee99ffb2082e13e95ee285f32fb2e599765b08b4404d8e475bacff79a412a954d227b93ba53ef
+    REF 0fb05a47528559c8964d681e2b92815c5559b10d
```
Comment (NancyLi1013) [inline ports/opentracing/portfile.cmake]: Why do you update this port here? Could you please make a simple description about this?

**Comment by NancyLi1013** [comment]
As for the issue on arm64-windows, I will have a further investigate.
For the change in ci.baseline.txt, you need to pull from vcpkg master branch to your branch. Since this file is added recently. If you can confirm this port currently doesn't support arm64, you can update it.

**Comment by xgcssch** [comment]
As stated in my initial comment to this pull request:

> This pull request contains an version upgrade to the opentracing port, as the previous version produced a release build as the debug version. This makes any code using the debug version to crash.

**Comment by NancyLi1013** [comment]
Thanks again.
Could you update the two lines as `vcpkg_fail_port_install(ON_ARCH "arm64" ON_TARGET "UWP")`?
```
vcpkg_fail_port_install(ON_TARGET "UWP")
vcpkg_fail_port_install(ON_ARCH "arm64")
```
Sorry, I didn't see this before. 
Besides this, it looks good to me now.

**Review by NancyLi1013 — APPROVED** [review]

**Comment by NancyLi1013** [comment]
Hi @xgcssch 
Could you please resolve the conflicts?

**Comment by xgcssch** [comment]
Hello @NancyLi1013 
I'm sorry, but it is not possible to resolve the conflict anymore.
Very annoying: due to the long delay, the time window for integration has passed:
the upstream project https://github.com/jaegertracing/jaeger-client-cpp is still using the Version 1.5 of opentracing. They have to migrate to Version 1.6 until we can give this new port another try.
Best regards 
Sönke

**Comment by NancyLi1013** [comment]
Hi @xgcssch 
Sorry for the late reply.
I try to update `opentracing `to 1.6.0 and then build the port with the following errors:
```
F:\9126\vcpkg\buildtrees\jaeger-client-cpp\src\cc1c5a2d08-596f1d4c7b\src\jaegertracing/Span.h(78): error C2259: 'jaegertracing::Span': cannot instantiate abstract class
F:\9126\vcpkg\buildtrees\jaeger-client-cpp\src\cc1c5a2d08-596f1d4c7b\src\jaegertracing/Span.h(36): note: see declaration of 'jaegertracing::Span'
F:\9126\vcpkg\buildtrees\jaeger-client-cpp\src\cc1c5a2d08-596f1d4c7b\src\jaegertracing/Span.h(78): note: due to following members:
F:\9126\vcpkg\buildtrees\jaeger-client-cpp\src\cc1c5a2d08-596f1d4c7b\src\jaegertracing/Span.h(78): note: 'void opentracing::v3::Span::Log(opentracing::v3::SystemTime,const std::vector<std::pair<opentracing::v3::string_view,opentracing::v3::Value>,std::allocator<std::pair<opentracing::v3::string_view,opentracing::v3::Value>>> &) noexcept': is abstract
F:\9126\vcpkg\installed\x86-windows\include\opentracing/span.h(181): note: see declaration of 'opentracing::v3::Span::Log'
F:\9126\vcpkg\buildtrees\jaeger-client-cpp\src\cc1c5a2d08-596f1d4c7b\src\jaegertracing/Span.h(78): note: 'void opentracing::v3::Span::Log(opentracing::v3::SystemTime,std::initializer_list<std::pair<opentracing::v3::string_view,opentracing::v3::Value>>) noexcept': is abstract
F:\9126\vcpkg\installed\x86-windows\include\opentracing/span.h(177): note: see declaration of 'opentracing::v3::Span::Log'
[7/185] C:\PROGRA~2\MICROS~3\2019\COMMUN~1\VC\Tools\MSVC\1425~1.286\bin\Hostx64\x86\cl.exe   /TP -DBOOST_CONFIG_SUPPRESS_OUTDATED_MESSAGE -DYAML_CPP_DLL -D_SCL_SECURE_NO_WARNINGS -D_SILENCE_CXX17_ITERATOR_BASE_CLASS_DEPRECATION_WARNING -Djaegertracing_EXPORTS -IF:\9126\vcpkg\buildtrees\jaeger-client-cpp\src\cc1c5a2d08-596f1d4c7b\src -Isrc -IF:\9126\vcpkg\installed\x86-windows\include /DWIN32 /D_WINDOWS /W3 /utf-8 /GR /EHsc /MP  /D_DEBUG /MDd /Z7 /Ob0 /Od /RTC1 /showIncludes /FoCMakeFiles\jaegertracing.dir\src\jaegertracing\Tag.cpp.obj /FdCMakeFiles\jaegertracing.dir\ /FS -c F:\9126\vcpkg\buildtrees\jaeger-client-cpp\src\cc1c5a2d08-596f1d4c7b\src\jaegertracing\Tag.cpp
Microsoft (R) C/C++ Optimizing Compiler Version 19.25.28610.4 for x86
```
Since `opentracing `has been update to the latest version 1.6.0 in vcpkg. 
Currently, we should support the new version.
So could you please post this issue to `jaeger-client-cpp` and let them look into it?

**Comment by xgcssch** [comment]
Rebased with head and removed the change for opentracing.
This should build now

**Comment by NancyLi1013** [comment]
It seems that it doesn't work in current state.
Could you please take a look?

**Comment by xgcssch** [comment]
Sorry for the extra iteration. My change didn't made it into the repo.
Cheers

File: ports/jaeger-client-cpp/CONTROL [hunk 4]
```diff
@@ -0,0 +1,5 @@
+Source: jaeger-client-cpp
+Version: 0.5.1
+Build-Depends: nlohmann-json,thrift,yaml-cpp,opentracing
+Homepage: https://github.com/jaegertracing/jaeger-client-cpp
+Description: C++ OpenTracing binding for Jaeger https://jaegertracing.io/
```
Comment (NancyLi1013) [inline ports/jaeger-client-cpp/CONTROL:5]: Since this port currently cannot build on `arm64-windows`.
Can we `add Supports: !arm64` here?

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]
Could you help confirm if the port doesn't support uwp? or `opentracing `doesn't support uwp, which caused that `jaeger-client-cpp` cannot build on uwp?
If it is due to `opentracing `, we should remove uwp from` vcpkg_fail_port_install()` in portfile.cmake file.

I'm sorry. I didn't notice this before.

**Comment by NancyLi1013** [comment]
Thanks for your continuous efforts on this PR. All pipelines build successfully now except for osx.
As for the regression on osx pipeline, we are trying to fix this.
Once passed, I will re run this PR.

File: ports/jaeger-client-cpp/CONTROL (hunk 4 above)
Comment (xgcssch) [reply ports/jaeger-client-cpp/CONTROL:5]: I tested the compile with x64:uwp =>
The dependency `nlohmann-json` has a build error there.
So i can't say if the port will build.

For the supposed changes:
Do you mean => change
`vcpkg_fail_port_install(ON_ARCH "arm64" ON_TARGET "UWP")`
to
`vcpkg_fail_port_install(ON_ARCH "arm64")`

and add a line to CONTROL like
`Supports: !arm64`
BTW: should't it be like
`Supports: !(uwp|arm)` ?
Sönke

Comment (NancyLi1013) [reply ports/jaeger-client-cpp/CONTROL:5]: Since it is blocked by `opentracing `on uwp.
So we only need to add `arm64`.

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

**Review by NancyLi1013 — APPROVED** [review]

**Comment by strega-nil** [comment]
Cool, this looks great! Thanks @xgcssch :)

**Pull request merged**